### PR TITLE
feat: 🎸 Post about Side-Projects

### PR DESCRIPTION
**Pull request by dirtyhenry**

### CODE REVIEW

File: _posts/2022-12-12-side-projects-2022.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+---
+layout: post
+id: E629FACC-8499-4402-B03B-0F5DB13396BD
+title: The Sparking Joy of Side-Projects, 2022 Edition
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:4]: The Joy Sparked by Side Projects

File: _posts/2022-12-12-side-projects-2022.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+author: Mick F
+excerpt: >-
+  Reviewing 2022, the benefits I gained from running a newsletter as a
+  side-project are tremendous. Let me explain why.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:8]: I think you can just write "side project" without a hyphen

File: _posts/2022-12-12-side-projects-2022.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+category: Journaling
+tags:
+  - Swift
+  - iOS
+  - JavaScript
+  - React
+---
+
+🎄 December is here! Like every year, it is time to watch _Home Alone_ again and
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:17]:
```suggestion
🎄 December is here! Like every December, it is time to watch _Home Alone_ again and
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+for a review of the progress I made as a developer in the past 12 months.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:18]:
```suggestion
to review the progress I made as a developer over the past 12 months.
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+Side-projects contributed the most.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:19]:
```suggestion
Side projects contributed the most.
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+## Why side-projects are necessary to me
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:21]:
```suggestion
## Why side projects are necessary for me
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+Yes, a paying job is your bread and butter. But having a side-project is key to
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:23]:
```suggestion
Yes, a paying job is your bread and butter. But having a side project is key to
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+stay ahead of the younger guns of the tech world, flexing your developer muscles
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:24]:
```suggestion
staying ahead of the younger guns of the tech world, flexing your developer muscles
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+in new directions and anticipating what’s next for you at work.
+
+Last year, I started writing [a newsletter about soccer/football][1][^1] for
+people based in France. The content is pretty basic. Still, I learnt so much
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:28]:
```suggestion
people based in France. The content is pretty basic. Still, I learned so much
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+just setting it up, running it, and trying to automate as many steps as possible
+so that each edition would take as little time as possible to write, without
+compromising on its value.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:31]:
```suggestion
compromising the quality of the content.
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+## Things I learned
+
+Here is a non-exhaustive list of the things I trained on last year to run that
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:35]:
```suggestion
Here is a non-exhaustive list of the things I trained on last year to run the
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+newsletter, in an approximative chronological order:
+
+- [**Swift Argument Parser.**][3] Writing a CLI in Swift is now my goto
+  direction when I need to write scripts. Including at work. It is so much
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:39]:
```suggestion
  solution when I need to write scripts. Including at work. It is so much
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+  easier to write, document and maintain than Shell scripts.
+- [**Buttondown.**][4] This is an indie tool to bootstrap newsletters. As part
+  of the side-project journey, I always thought it was worth exploring
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:42]:
```suggestion
  of the side project journey, I believe it is worth exploring
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+  alternatives to big corporations so I skipped Mailchimp, Mailjet and similar
+  options. While I really like it, I think the newsletter is now ready to grow
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:44]:
```suggestion
  options. While I really like Buttondown, I think the newsletter is now ready to grow
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+  out of it, and finding a better option for my needs is on the roadmap
+  for 2023.
+- [**Linode.**][5] Similarly, I didn’t want to go the AWS route. Since I
+  started, Linode got bought by Akamai so I am not as attached to it as I was.
+  Much clearer than AWS still.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:49]: what does this mean? You still prefer it?

File: _posts/2022-12-12-side-projects-2022.md [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+- **Modern React web frameworks.** I explored [Next.js][11] first, then
+  [Remix][12]. Both are pretty amazing.
+- [**DocC.**][13] The new documentation framework by Apple. I used [Jazzy][20]
+  before, but DocC — albeit sometimes awkward to use — is much more complete. As
+  illustrated by [Simon Støvring's talk during FrenchKit 2022][i1], you can
+  create amazing documentation with it.
+- [**Xcode Cloud.**][14] First-party CI and automated distribution of apps. It
+  seems to cover some common ground with Fastlane, very convenient to distribute
+  the app I use to run the newsletter on my iPhone and iPad via TestFlight.
+- [**TabularData.**][15] A new framework provided by Apple to organize, and
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:69]:
```suggestion
- [**TabularData.**][15] A new framework provided by Apple to organize and
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+
+Each of these tools would deserve their own post. I shouldn't run out of
+inspiration for this blog next year.
+
+## The positive impacts on my paying job
+
+### 🧰 A better set of tools I can use
+
+Part of my job at Memo Bank is to write the iOS app that serves as the 2FA for
+critical operations. It does not do much. But the things it does are critical.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:87]:
```suggestion
critical operations. The app does not do much. But the things it does are critical.
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 18]
```diff
@@ -0,0 +1,172 @@
...
+Our customers depend on it to pay salaries to their employees, pay taxes in
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:88]:
```suggestion
Our customers depend on it to pay salaries to their employees, pay taxes on
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 19]
```diff
@@ -0,0 +1,172 @@
...
+time, and be safe with their transactions.
+
+The app was boostrapped with a compatibility going up to iOS 12. Now that I got
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:91]:
```suggestion
The app was bootstrapped with compatibility going up to iOS 12. Now that I got
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+my hands dirty with SwiftUI and _The Composable Architecture_ on my
+side-projects, I feel comfortable adopting them in the app for upcoming
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:93]:
```suggestion
side projects, I feel comfortable adopting them in the app for upcoming
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+features. That would not have been the case without the time I spent toying with
+them. I made many mistakes. I rewrote things multiple times. It is part of
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:95]:
```suggestion
them outside of work. I made many mistakes. I rewrote things multiple times. It is part of
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 22]
```diff
@@ -0,0 +1,172 @@
...
+learning.
+
+Could this time spent on my free time have been part of the job? Maybe. But I
+also like not having to be held accountable for anything when I toy with things.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:99]:
```suggestion
also like not having to be held accountable for anything when I explore and want to play around with new things.
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 23]
```diff
@@ -0,0 +1,172 @@
...
+## My tips and life lessons
+
+**🧱 Use a monorepo.** Over time, I realized that using a monorepo for my
+projects was much more thrilling than having a million bunny projects with 5
+commits each. The feeling of progress is much more satisfying.
+
+**🏎 Take shortcuts towards the fun.** If you host your code on GitHub, don't use
+an organization and don't open-source what you are doing until you feel ready.
+For instance, at work, you should never leave secrets in your code. On a
+side-project, why not? If stakes are low, leave the boring requirerd things you
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:127]:
```suggestion
side project, why not? If stakes are low, leave the boring required things you
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 24]
```diff
@@ -0,0 +1,172 @@
...
+write at work behind and sprint towards the fun parts.
+
+**⏱ How to find time for it?** Any opportunity is good to take. My wife agreed
+to leave me off kids-duty on Thursday evenings. I try to write a little bit of
+code everyday, and when you work remotely, the lunch break is ideal for that.
+Over time, things build up and the output you get from a 5 minutes slot grows.
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:133]:
```suggestion
Over time, things build up and the output you get from a 5-minute slot grows.
```

File: _posts/2022-12-12-side-projects-2022.md [hunk 25]
```diff
@@ -0,0 +1,172 @@
...
+
+Also, I recommend reading [these tips from Antoine van der Lee][19]. They are
+great to act as a guard rail against uncontrollable impulses to try this new
+thing. His point is to keep focus, realistic expecations and priorities
+straight.
+
+## Conclusion
+
+Why are you still reading this? Go work on your side-project!
```
Comment (alkidel) [inline _posts/2022-12-12-side-projects-2022.md:142]:
```suggestion
Why are you still reading this? Go work on your side project!
```

File: _posts/2022-12-12-side-projects-2022.md (hunk 3 above)
Comment (dirtyhenry) [reply _posts/2022-12-12-side-projects-2022.md:17]: I didn't want to repeat `December`. What is wrong with `year`? What else can I say to not repeat `December`?

File: _posts/2022-12-12-side-projects-2022.md (hunk 15 above)
Comment (dirtyhenry) [reply _posts/2022-12-12-side-projects-2022.md:49]: I guess so. Yay, I'll try to rephrase to make it clearer.

**Pull request merged**